MovieCave - MVVM/MovieCave/MovieCave/Coordinators/MoviesViewCoordinator.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MoviesViewCoordinator.swift
+//  MovieCave
+//
+//  Created by Admin on 28.09.23.
+//
+
+import UIKit
+
+protocol MoviesViewCoordinatorDelegate: AnyObject {
+    
+    /// Loads the MovieDetailsView and ViewModel to display additional
+    /// information for a selected movie.
+    /// - Parameter ID: The ID of the selected movie to load details for.
+    func loadMoviesDetailsView(with movieID: Int)
+    
+    /// Removes the coordinator from coordinator  hierarchy.
+    func delocateCoordinator()
+}
+
+class MoviesViewCoordinator: Coordinator, MoviesViewCoordinatorDelegate {
+    
+    //MARK: - Properties
+    private var navController: UINavigationController
+    private var list: MoviesList
+    
+    //MARK: - Initialization
+    init(navController: UINavigationController, with list: MoviesList) {
+        self.navController = navController
+        self.list = list
+    }
+    
+    //MARK: - Override methods
+    override func start() {
+        guard let moviesVC = MoviesViewController.initFromStoryBoard() else { return }
+
+        moviesVC.viewModel = MoviesViewModel(coordinator: self,
+                                             movieDBService: MovieDBService(),
+                                             dataSource: CollectionViewDataSource(items: []),
+                                             with: list,
+                                             with: .topRated,
+                                             currentPage: 1)
```

Comment:
Try to avoid 'magic' numbers in the code. Extract to a constant somewhere in this class/file/etc, it's up to you where